spec/models/video_spec.rb

```diff
@@ -35,4 +36,25 @@
       expect(Video.search_by_title "").to eq([])
     end
   end
+
+  describe "#calculate_rating" do
+    let(:video) { Fabricate(:video) }
+
+    it "returns 0 if there are no reviews" do
+      expect(video.calculate_rating).to eq(0.0)
+    end
+
+    it "returns a single integer value if there is one review" do
+      review = Fabricate(:review, video: video)
+      expect(video.calculate_rating).to eq(review.rating)
+    end
+
+    it "returns an average of all the ratings if there is more than one review" do
+      review1 = Fabricate(:review, video: video)
+      review2 = Fabricate(:review, video: video)
+      review3 = Fabricate(:review, video: video)
+      average = (review1.rating + review2.rating + review3.rating) / 3.0
```

Comment:
You can also be more explicit and hard code your ratings in your fabrications. This code depends on your fabricator being setup correctly. If you hard code the ratings, you can also hard code the expected value.